ui/imports/shared/views/HistoryView.qml

```diff
@@ -7,21 +7,22 @@ import utils 1.0
 import StatusQ.Components 0.1
 import StatusQ.Controls 0.1
 
-import shared.panels 1.0
+import "../panels"
 import "../popups"
 import "../stores"
 import "../controls"
 
 Item {
     id: historyView
 
+    property var account
     property int pageSize: 20 // number of transactions per page
 
     function fetchHistory() {
-        if (RootStore.isFetchingHistory()) {
+        if (RootStore.isFetchingHistory(historyView.account.address)) {
             loadingImg.active = true
         } else {
-            RootStore.loadTransactionsForAccount(pageSize)
+            RootStore.loadTransactionsForAccount(historyView.account.address, pageSize)
```

Comment:
how is this resolved? in case of wallet and browser?

RootStore was a singelton qml under Wallet and Browser, when this is placed under shared I am not sure how it resolves this?